Live/LiveEventWithDVR/Program.cs

```diff
@@ -79,522 +74,375 @@
+PrintIngestUrls(liveEvent);
 
-    // Note: When creating a LiveEvent, you can specify allowed IP addresses in one of the following formats:                 
-    //      IpV4 address with 4 numbers
-    //      CIDR address range  
+var streamingLocator = await DemoStreamingAsync(
+    mediaServicesAccount,
+    asset,
+    streamingEndpointName,
+    streamingLocatorName,
+    manifestName,
+    dvrAssetFilterName);
 
-    var allAllowIpRange = new IPRange()
-    {
-        Name = "AllowAll",
-        Address = IPAddress.Parse("0.0.0.0"),
-        SubnetPrefixLength = 0
-    };
+Console.WriteLine("Cleaning up resources, stopping Live Event billing, and deleting live Event...");
+await CleanupResourcesAsync(
+    liveEvent,
+    liveOutput,
+    streamingLocator,
+    asset);
 
-    // Create the LiveEvent Preview IP access control object. 
-    // This will restrict which clients can view the preview endpoint
-    var liveEventPreview = new LiveEventPreview();
-    liveEventPreview.IPAllowedIPs.Add(allAllowIpRange);
+if (mediaServicesMonitor != null)
+{
+    await mediaServicesMonitor.StopAsync();
+}
+
+static async Task<MediaLiveEventResource> CreateLiveEventAsync(
+    MediaServicesAccountResource mediaServicesAccount,
+    string liveEventName)
+{
+    // Creating the LiveEvent - the primary object for live streaming in Media Services.
+    // See the overview - https://learn.microsoft.com/azure/media-services/latest/live-event-concept
 
-    #region NewLiveEvent
     var liveEventData = new MediaLiveEventData(mediaServicesAccount.Get().Value.Data.Location)
     {
-        Description = "Sample LiveEvent from .NET SDK sample",
+        Description = "Sample LiveEvent from the .NET SDK sample",
         UseStaticHostname = true,
         // 1) Set up the input settings for the Live event...
         Input = new LiveEventInput(streamingProtocol: LiveEventInputProtocol.Rtmp)
         {
             StreamingProtocol = LiveEventInputProtocol.Rtmp,
             AccessToken = "acf7b6ef-8a37-425f-b8fc-51c2d6a5a86a",
-            KeyFrameIntervalDuration = TimeSpan.FromSeconds(2)
+            KeyFrameIntervalDuration = TimeSpan.FromSeconds(2),
+            IPAllowedIPs =
+            {
+                new IPRange()
+                {
+                    Name = "AllowAllIpV4Addresses",
+                    Address = IPAddress.Parse("0.0.0.0"),
+                    SubnetPrefixLength = 0
+                },
+                new IPRange()
+                {
+                    Name = "AllowAllIpV6Addresses",
+                    Address = IPAddress.Parse("0::"),
+                    SubnetPrefixLength = 0
+                }
+            }
         },
         // 2) Set the live event to use pass-through or cloud encoding modes...
         Encoding = new LiveEventEncoding()
         {
             EncodingType = LiveEventEncodingType.PassthroughStandard
```

Comment:
Suggest using Basic Pass through now which is cheaper, or switch to 720p transcoding which is more common.

Reply:
Updated.